installer/lib/mix/tasks/alice_new_handler.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Mix.Tasks.Alice.New.Handler do
+  @moduledoc """
+  Generates a new Alice handler.
+
+  It expects the path of the handler as an argument.
+
+      mix alice.new.handler PATH [--module MODULE] [--app APP]
+
+  An Alice handler at the given PATH will be created.
+  The application name and module name will be retrieved
+  from the path, unless `--module` or `--app` is given.
+
+  ## Options
+
+    * `--app` - the name of the OTP application
+
+    * `--module` - the name of the handler module
+
+  ## Examples
+
+      mix alice.new.handler hello_world
+
+  Is equivalent to:
+
+      mix alice.new.handler hello_world --module HelloWorld --app hello_world
+
+  To print the Alice installer version, pass `-v` or `--version`, for example:
+
+      mix alice.new.handler -v
+
+  To print this help text, pass `-h` or `--help`, for example:
+
+      mix alice.new.handler -h
+  """
+  use Mix.Task
+
+  alias AliceNew.{
+    HandlerGenerator,
+    Utilities
+  }
+
+  @shortdoc "Creates a new Alice v#{Utilities.alice_version()} handler"
+
+  @switches [
+    app: :string,
+    module: :string
+  ]
+
+  def run([version]) when version in ~w[-v --version] do
+    Mix.shell().info("Alice v#{Utilities.alice_version()}")
+  end
+
+  def run([help]) when help in ~w[-h --help] do
+    Mix.Tasks.Help.run(["alice.new.handler"])
+  end
+
+  def run(argv) do
+    case parse_opts(argv) do
+      {_opts, []} ->
+        Mix.Tasks.Help.run(["alice.new.handler"])
+
+      {opts, [path | _]} ->
+        Utilities.elixir_version_check!()
+
+        basename = Path.basename(Path.expand(path))
+        path = Path.join([Path.dirname(path), "alice_#{basename}"])
+
+        handler_name = opts[:app] || basename
+        app = "alice_#{handler_name}"
+        Utilities.check_handler_name!(handler_name, !opts[:app])
+
+        module_name = opts[:module] || Macro.camelize(handler_name)
+        Utilities.check_mod_name_validity!(module_name)
+        module = Utilities.handler_module(module_name)
```

Comment:
For some reason I think this might be better encapsulated in a single function in Utilities. Something like:
```elixir
def generate_handler_module(module_name) do
  module_name
  |> check_mod_name_validity!
  |> handler_module
end
```
That way on this side, you're just generating variables and then passing them into a single function that does both the verification and the transformation.

Reply:
I think Utilities is a bad name, it's not meant to be a general assortment of functions. Maybe I should have made them all private functions in this module for now, but the idea was to put functions there that aren't specific to the handler task.

The next step for the installer is making a bot generator, so the idea was to put common functions in Utilities, but since we don't have that yet, it makes the line hard to draw. I guess this makes the module yagni.

Aside from that, I don't always think it's necessary to extract a function unless it's going to be used in more than one place or unless the code is confusing and needs a name. In this case, this might be better than extracting a function:

```suggestion
        module_name = opts[:module] || Macro.camelize(handler_name)

        module =
          module_name
          |> Utilities.check_mod_name_validity!()
          |> Utilities.handler_module()
```

Reply:
That’s certainly better than it was before, thought DRY is usually the secondary reason for extraction IMO. It’s usually to put a meaningful name on it as the primary reason.

Reply:
The primary purpose of extraction is abstraction. Or as Rich Hickey would put it. “To decomplect” 😄